docs/premium-features/custom-types-and-actions.md

```diff
@@ -0,0 +1,259 @@
+
+Amplication provides built-in DTOs for common use cases, such as pagination, sorting, and filtering. These DTOs are automatically generated based on your application's entities to support the CRUD APIs generated out-of-the-box. However, you can also create custom DTOs to suit your specific requirements and use them in your custom APIs.
+
+### Differences between DTOs and Enums
+
+Enums are used to define a fixed set of constants. DTOs are used to define the structure and shape of data objects that are transferred between different parts of the project.
+
+In Amplication, you can create and manage both enums and DTOs using the Custom Types & Actions feature, allowing you to define the data contracts and constraints for your application's API endpoints. A specific enum can also function as a DTO and can be defined as the input or output DTO for an API.
+
+### Create DTO
+
+To create a new DTO, follow these steps:
+
+:::note
+If you create a new DTO / Enum from the "All Modules" screen, you can choose which Module it will reside in.
+:::
+
+#### DTO Name and Description
+
+1. Click on the "Add DTO" button in the top right of the APIs tab.
+2. In the dropdown, click on "Add DTO".
+3. Provide a name for your DTO, such as "MyNewDtoName". This will automatically populate the Name field with the provided name.
+4. Optionally, add a description for your DTO.
+
+#### Adding Properties
+
+1. In the "Properties" section, click on "Add property" to define the DTO's properties.
+2. For each property:
+   - Specify the Name of the property (e.g., "firstProperty").
+   - Choose the Type of the property from the dropdown (e.g., "String", "Number", "Boolean", "Date", "JSON", "DTO", "Enum").
+   - If needed, you can change the property to an array by toggling the "Array" switch.
+   - Mark the property as optional by checking the "Optional" checkbox.
+3. Repeat step 2 for each additional property you want to add to the DTO.
+
+### Create Enum
+
+To create a new Enum, follow these steps:
+
+#### Enum Name and Description
+
+1. Click on the "Add DTO" button in the top right of the APIs tab.
+2. In the dropdown, click on "Add Enum".
+3. Provide a name for your Enum, such as "MyNewEnum". This will automatically populate the Name field with the provided name.
+4. Optionally, add a description for your Enum.
+
+#### Adding Members
+
+1. In the "Members" section, click on "Add member" to define the Enum's members.
+2. For each member:
+   - Specify the Name of the member.
+   - Provide the Value for the member.
+3. Repeat step 2 for each additional member you want to add to the Enum.
+
+### Edit DTO or Enum
+
+To edit an existing DTO, click on the DTO within its associated Module in the APIs tab. You can modify the DTO's name, description, and properties as needed.
+
+## Actions
+
+Actions represent the API endpoints that define the operations available in your application. They specify the HTTP methods, paths, input, and output data structures for each endpoint.
```

Comment:
This text is more REST-centric than GQL centric. Not sure it's a problem, but a point to consider.
Anyway, if we leave it as is, let's change methods to verbs and paths to routes